client/settings/settings.jsx

```diff
@@ -94,10 +94,6 @@ class SettingsForm extends React.Component {
           disabled={!this.isFullscreen()}
           inputProps={{ tabIndex: 0 }}
         />
-        <Select {...bindCustom('renderer')} label='Renderer' tabIndex={0}>
```

Comment:
I think we should leave this select in the settings for now, and just remove the `OpenGL` option from it. I know that having a select with only one option is a bit weird, but this whole settings dialog will need to be revamped pretty soon depending on what we decide to do with regards to SC:R support.

You can also remove any OpenGL specific code that we had in forge, eg.
https://github.com/ShieldBattery/ShieldBattery/blob/rust-dll/game/rust-dll/src/forge/renderer.rs#L26

Another thing that you should take care of is handle the case for users that have OpenGL saved in their settings. Guess it's already handled if you ignore that setting in forge and are just creating the DirectX renderer? We also need a migration to update those user's settings file, but I guess we can postpone that for when we revamp the settings.

Reply:
I feel that leaving the select as something that just sits there saying 'This is not something you can change' is weird, but sure, I don't have strong preference either way.

And right, forge can be cleaned up as well - I was thinking if it's worth saving the details on what kind of hacks opengl needs to do in order to get acceptable rendering, but I also think it still didn't work properly all the time with Windows 10's taskbar?

Reply:
Yeah, I'm not sure either which is better; completely remove the renderer option for now, or just remove the OpenGL option from it. If we go with former option, then we should also publish the new version of settings and add a migration for it here:
https://github.com/ShieldBattery/ShieldBattery/blob/master/app/local-settings.js#L98

Btw, I can't really figure out how the settings are handled now in the Rust code, but the game code should handle the case of undefined/invalid setting values and apply the default values, where applicable, ie. what was being done here:
https://github.com/ShieldBattery/ShieldBattery/blob/master/game/node-bw/wrapped_brood_war.cpp#L467

And as far as the OpenGL hacks go, I can't really remember what was the state of things there; but I don't think we get any benefit of keeping those hacks around if we're not planning on implementing OpenGL renderer, especially if we move to SC:R (in any case, there's always git history that we can turn to if we really need to dig those in future :D).

Reply:
The settings are technically handled here: https://github.com/ShieldBattery/ShieldBattery/blob/328b57c035bd4dff0ecd8a0ae1577a4a13e5d310/game/rust-dll/src/game_state.rs#L223
but since it turned out that all settings we currently have are only relevant to forge, it currently doesn't do anything else than forward to forge which reads what it needs https://github.com/ShieldBattery/ShieldBattery/blob/328b57c035bd4dff0ecd8a0ae1577a4a13e5d310/game/rust-dll/src/forge/mod.rs#L1123
(Invalid/undefined values are handled same as the c++ did code except it doesn't read the renderer field)

Reply:
Oh, I missed that; that's cool.

Reply:
I removed the renderer/opengl from everywhere and added settings migration for new version; it would not be useful with SCR anyway (Or rather adding custom renderer for SCR would be a completely new, unrelated and huge task)